web-modules/javax-servlets-2/src/main/java/com/baeldung/servlets/FormDataServlet.java

```diff
@@ -0,0 +1,24 @@
+package com.baeldung.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "FormDataServlet", urlPatterns = "/form-data")
+public class FormDataServlet extends HttpServlet {
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        String firstName = req.getParameter("first_name");
+        String lastName = req.getParameter("last_name");
+
+        resp.getWriter()
+            .append("Full Name: ")
+            .append(firstName)
```

Comment:
This is, of course, vulnerable to XSS. So please again either point to the OWASP reference or use something more robust.

Reply:
I think it would be helpful to mark it here in the code as well. Something like:

```
// Don't use this code as-is! It is vulnerable to XSS. 
// See 
// https://www.baeldung.com/cs/cross-site-scripting-xss-explained and
// https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html
// to learn how to harden it.
```